habitat-hitl/README.md

```diff
@@ -41,29 +45,84 @@ Example HITL apps are configured to run at 30 steps per second (SPS). If your sy
+The HITL framework can be used to build desktop applications (controlled with keyboard/mouse) as well as **VR** applications. See our [Pick_throw_vr](../examples/hitl/pick_throw_vr/README.md) example application.
 
 ## Configuration
 
-HITL apps use Hydra for configuration, for example, to control the desktop window width and height. See [`hitl_defaults.yaml`](./config/hitl_defaults.yaml) as well as each example app's individual config, e.g. [`pick_throw_vr.yaml`](../hitl/pick_throw_vr/config/pick_throw_vr.yaml). See also [`habitat-lab/habitat/config/README.md`](../../habitat-lab/habitat/config/README.md).
-
-## HITL Framework Architecture
-* The HITL framework is logically divided into a Habitat-lab environment wrapper (`SandboxDriver`) and a GUI component (`GuiApplication` and `ReplayGuiAppRenderer`).
-* `SandboxDriver`
-    * It creates a `habitat.Env` instance.
-    * Camera sensors are rendered by the `habitat.Env` instance in the usual way; see `self.obs = self.env.step(action)` in `SandboxDriver.sim_update`.
-    * This class is provided a `gui_input` object that encapsulates OS input (keyboard and mouse input). We should avoid making direct calls to PyGame, GLFW, and other OS-specific APIs.
-    * `sim_update` returns a `post_sim_update_dict` that contains info needed by the app renderer (below). E.g. a gfx-replay keyframe and a camera transform for rendering, plus optional "debug images" to be shown to the user.
-    * This class also has access to a `debug_line_render` instance for visualizing lines in the GUI (the lines aren't rendered into camera sensors). This access is somewhat hacky; future versions of HITL apps will likely convey lines via `post_sim_update_dict` instead of getting direct access to this object.
-    * This class is provided an AppState which provides application-specific logic, for example, specific keyboard/mouse controls and specific on-screen help text.
-* `GuiApplication`
-    * manages the OS window (via GLFW for now), including OS-input-handling (keyboard and mouse) and updating the display (invoking the renderer).
-* `ReplayGuiAppRenderer`
-    * `ReplayGuiAppRenderer` is a "render client". It receives the `post_sim_update_dict` from `SandboxDriver` and updates the OS window by rendering the scene from the requested camera pose.
-    * In theory, render clients should be app-agnostic, i.e. `ReplayGuiAppRenderer` could be re-used for other GUI apps, but in practice we may find situations where we have to inject some app-specific code into this class (we should avoid if possible).
+HITL apps use Hydra for configuration, for example, to control the desktop window width and height. See [`hitl_defaults.yaml`](./config/hitl_defaults.yaml) as well as each example app's individual config, e.g. [`pick_throw_vr.yaml`](../examples/hitl/pick_throw_vr/config/pick_throw_vr.yaml). See also [`habitat-lab/habitat/config/README.md`](../habitat-lab/habitat/config/README.md).
+
+## Minimal HITL application
```

Comment:
We may want to refer to `minimal.py` instead of putting code here so that we don't have a maintenance overheard/regression.

Reply:
I am very interested to have source code snippet in the README, just to emphasize to people how little code is required. Agree it'll be a maintenance headache. I will add a note at the bottom of the snippet to encourage people to click through to the source file for the latest version.